Type checking does't work

### Environment

- Operating System: `Windows_NT`
- Node Version:     `v16.14.0`
- Nuxt Version:     `3.0.0-rc.3`
- Package Manager:  `npm@8.3.1`
- Builder:          `vite`
- User Config:      `telemetry`, `modules`, `typescript`, `runtimeConfig`, `css`
- Runtime Modules:  `trpc-nuxt@0.1.7`
- Build Modules:    `-`


### Reproduction

https://stackblitz.com/edit/github-qruhrz?file=app.vue

### Describe the bug

When you `typescript.typeCheck: true`, type errors are still ignored. This happens both in the build and dev environments.

### Additional context

_No response_

### Logs

_No response_

Also typecheck with `nuxi typecheck` throws an error:

```
Nuxt CLI v3.0.0-rc.3
npm ERR! Cannot read properties of undefined (reading 'replace')

npm ERR! A complete log of this run can be found in:
npm ERR!     /Somewhere/Someone/.npm/_logs/2022-05-23T15_04_25_555Z-debug-0.log

 ERROR  Command failed with exit code 1: npx -p vue-tsc -p typescript vue-tsc --noEmit

  at makeError (node_modules/nuxi/dist/chunks/index5.mjs:1242:11)
  at handlePromise (node_modules/nuxi/dist/chunks/index5.mjs:2071:26)
  at processTicksAndRejections (node:internal/process/task_queues:96:5)
  at async Object.invoke (node_modules/nuxi/dist/chunks/typecheck.mjs:43:7)
  at async _main (node_modules/nuxi/dist/cli.mjs:46:20)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I don't know if this is related. But it happens. 😓

--- a/packages/vite/src/server.ts
+++ b/packages/vite/src/server.ts
@@ -88,7 +88,7 @@ export async function buildServer (ctx: ViteBuildContext) {
   // Add type-checking
   if (ctx.nuxt.options.typescript.typeCheck === true || (ctx.nuxt.options.typescript.typeCheck === 'build' && !ctx.nuxt.options.dev)) {
     const checker = await import('vite-plugin-checker').then(r => r.default)
-    ctx.config.plugins.push(checker({ typescript: true }))
+    serverConfig.plugins.push(checker({ typescript: true }))
   }
 
   await ctx.nuxt.callHook('vite:extendConfig', serverConfig, { isClient: false, isServer: true })

